app/models/orphan.rb

```diff
@@ -132,46 +119,89 @@ def sponsorship_changed!
 
 private
 
-  def sponsored_siblings_does_not_exceed_siblings_count
-   return if sponsored_minor_siblings_count.nil? || sponsored_minor_siblings_count == 0
-   if minor_siblings_count.nil? || sponsored_minor_siblings_count > minor_siblings_count
-     errors.add(:sponsored_minor_siblings_count, 'cannot exceed minor siblings count')
-   end
+  def set_defaults
+    return if persisted?
+    default_sponsorship_status_to_unsponsored
+    default_orphan_status_to_active
+    default_priority_to_normal
   end
 
-  def default_sponsorship_status_unsponsored
-    if new_record?
-      self.orphan_sponsorship_status ||= OrphanSponsorshipStatus.find_by_name 'Unsponsored'
-    end
+  def default_sponsorship_status_to_unsponsored
+    self.orphan_sponsorship_status ||= OrphanSponsorshipStatus.
+      find_by_name 'Unsponsored'
+  end
+
+  def default_orphan_status_to_active
+    self.orphan_status ||= OrphanStatus.find_by_name 'Active'
+  end
+
+  def default_priority_to_normal
+    self.priority ||= 'Normal'
+  end
+
+  def set_province_code
+    self.province_code ||= partner_province_code
+  end
+
+  def generate_osra_num
+    self.osra_num = "#{province_code}%05d" % sequential_id
   end
 
-  def default_orphan_status_active
-    if new_record?
-      self.orphan_status ||= OrphanStatus.find_by_name 'Active'
+  def sponsored_siblings_does_not_exceed_siblings_count
+    if sponsored_siblings_exceed_total?
```

Comment:
Why is this a function? Only has one line comparison and it never gets called in other part of the code?